packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1840,6 +1880,38 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
 
     return errors;
   }
+
+  private renderInstanceMaintenancePolicy(
+    maxHealthPercentage?: number,
+    minHealthPercentage?: number,
+  ): CfnAutoScalingGroup.InstanceMaintenancePolicyProperty | undefined {
+    if (maxHealthPercentage === undefined && minHealthPercentage === undefined) return;
+    if (maxHealthPercentage === undefined || minHealthPercentage === undefined) {
+      throw new Error(`Both or neither of maintenancePolicyMinHealthPercentage and maintenancePolicyMaxHealthPercentage must be specified, got maintenancePolicyMinHealthPercentage: ${minHealthPercentage} and maintenancePolicyMaxHealthPercentage: ${maxHealthPercentage}`);
+    }
+    if (maxHealthPercentage === -1 && minHealthPercentage === -1) {
+      return {
+        maxHealthyPercentage: maxHealthPercentage,
+        minHealthyPercentage: minHealthPercentage,
+      };
+    }
+    if (minHealthPercentage === -1 || maxHealthPercentage === -1) {
+      throw new Error(`Both maintenancePolicyMinHealthPercentage and maintenancePolicyMaxHealthPercentage must be -1 to clear the previously set value, got maintenancePolicyMinHealthPercentage: ${minHealthPercentage} and maintenancePolicyMaxHealthPercentage: ${maxHealthPercentage}`);
+    }
+    if (maxHealthPercentage < 100 || maxHealthPercentage > 200) {
+      throw new Error(`maintenancePolicyMaxHealthPercentage must be between 100 and 200, or -1 to clear the previously set value, got ${maxHealthPercentage}`);
+    }
+    if (minHealthPercentage < 0 || minHealthPercentage > 100) {
+      throw new Error(`maintenancePolicyMinHealthPercentage must be between 0 and 100, or -1 to clear the previously set value, got ${minHealthPercentage}`);
+    }
+    if (maxHealthPercentage - minHealthPercentage > 100) {
+      throw new Error(`The difference between maintenancePolicyMinHealthPercentage and maintenancePolicyMaxHealthPercentage cannot be greater than 100, got ${maxHealthPercentage - minHealthPercentage}`);
+    }
```

Comment:
We can store all errors we get in an array and throw them all at the same time, so the user will know all the issues present.

Reply:
Hmmm, is that necessary?

I think the information given to the user would be more complex, and type inference would not work and branching would become more complicated.

(Because of the eslint error, in numerical comparisons, `minHealthyPercentage !== undefined` and `maxHealthyPercentage !== undefined` must be added to the numerical comparison.)

```ts
    const errors: string[] = [];

    if (minHealthyPercentage === undefined && maxHealthyPercentage === undefined) return;
    if (minHealthyPercentage === undefined || maxHealthyPercentage === undefined) {
      errors.push(`Both or neither of minHealthyPercentage and maxHealthyPercentage must be specified, got minHealthyPercentage: ${minHealthyPercentage} and maxHealthyPercentage: ${maxHealthyPercentage}`);
    }
    if ((minHealthyPercentage === -1 || maxHealthyPercentage === -1) && minHealthyPercentage !== maxHealthyPercentage) {
      errors.push(`Both minHealthyPercentage and maxHealthyPercentage must be -1 to clear the previously set value, got minHealthyPercentage: ${minHealthyPercentage} and maxHealthyPercentage: ${maxHealthyPercentage}`);
    }
    if (minHealthyPercentage !== -1 && (minHealthyPercentage !== undefined && (minHealthyPercentage < 0 || minHealthyPercentage > 100))) {
      errors.push(`minHealthyPercentage must be between 0 and 100, or -1 to clear the previously set value, got ${minHealthyPercentage}`);
    }
    if (maxHealthyPercentage !== -1 && (maxHealthyPercentage !== undefined && (maxHealthyPercentage < 0 || maxHealthyPercentage > 100))) {
      errors.push(`maxHealthyPercentage must be between 100 and 200, or -1 to clear the previously set value, got ${maxHealthyPercentage}`);
    }
    if (maxHealthyPercentage !== undefined && minHealthyPercentage !== undefined && maxHealthyPercentage - minHealthyPercentage > 100) {
      errors.push(`The difference between minHealthyPercentage and maxHealthyPercentage cannot be greater than 100, got ${maxHealthyPercentage - minHealthyPercentage}`);
    }

    if (errors.length > 0) {
      throw new Error(errors.join('\n'));
    }

    return {
      minHealthyPercentage,
      maxHealthyPercentage,
    };
```